feature/experimental/gribi/otg_tests/programming_with_reload/programming_with_reload_test.go

```diff
@@ -0,0 +1,511 @@
+	maxRebootTime = 900
+	// Maximum gribi connection time is 180 seconds following reload
+	maxGribiConnectTime = 180
+)
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	dut    *ondatra.DUTDevice
+	ate    *ondatra.ATEDevice
+	top    gosnappi.Config
+	ctx    context.Context
+	client *gribi.Client
+}
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:1",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		MAC:     "02:00:01:01:01:01",
+		IPv4:    "192.0.2.2",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:2",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:5",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		MAC:     "02:00:02:01:01:01",
+		IPv4:    "192.0.2.6",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:6",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort3 = attrs.Attributes{
+		Desc:    "dutPort3",
+		IPv4:    "192.0.2.9",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:9",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort3 = attrs.Attributes{
+		Name:    "atePort3",
+		MAC:     "02:00:03:01:01:01",
+		IPv4:    "192.0.2.10",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:a",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePorts = map[string]attrs.Attributes{
+		"port1": atePort1,
+		"port2": atePort2,
+		"port3": atePort3,
+	}
+	dutPorts = map[string]attrs.Attributes{
+		"port1": dutPort1,
+		"port2": dutPort2,
+		"port3": dutPort3,
+	}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// configureATE configures ports on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
+	top := gosnappi.NewConfig()
+	for p, ap := range atePorts {
+		p1 := ate.Port(t, p)
+		dp := dutPorts[p]
+		ap.AddToOTG(top, p1, &dp)
+	}
+	return top
+}
+
+// configureDUT configures port1, port2, port3 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	for p, dp := range dutPorts {
+		p1 := dut.Port(t, p)
+		gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), dp.NewOCInterface(p1.Name(), dut))
+	}
+
+}
+
+// configureNetworkInstance configures vrfs vrfA, vrfB and add port1 to vrfA
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+	c := &oc.Root{}
+	vrfs := []string{vrfA, vrfB}
+	for _, vrf := range vrfs {
+		ni := c.GetOrCreateNetworkInstance(vrf)
+		ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+		if vrf == vrfA {
+			p1 := dut.Port(t, "port1")
+			niIntf := ni.GetOrCreateInterface(p1.Name())
+			niIntf.Subinterface = ygot.Uint32(0)
+			niIntf.Interface = ygot.String(p1.Name())
+		}
+		gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrf).Config(), ni)
+	}
```

Comment:
Please add below code to add other sub interfaces to  default vrf 
'''
	fptest.ConfigureDefaultNetworkInstance(t, dut)
	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
		fptest.AssignToNetworkInstance(t, dut, dut.Port(t, "port2").Name(), deviations.DefaultNetworkInstance(dut), 0)
		fptest.AssignToNetworkInstance(t, dut, dut.Port(t, "port3").Name(), deviations.DefaultNetworkInstance(dut), 0)
'''